feat(ZNTA-2070) editor settings options

kathryngo opened the pull request:
JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2070

Create stories for the Editor Options and Validation Settings using the SettingsOptions and SettingOption components.

Stories:

* SettingOption - default
* SettingsOptions - default - unchecked
* SettingsOptions -default - half-checked
* SettingsOptions - default - all checked
* EDITOR OPTIONS
* VALIDATION SETTINGS


## Reviewer tasks

The following is based on the
[Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines).
As a reviewer, you should check the guidelines regularly for updates, and just
to refresh your memory.


The reviewer can use this list for reference to make sure
they have considered all the important points.

Use the checkboxes or not, whatever works best for you.

 - Code quality
   - [ ] Code passes style check (checkstyle/eslint)
   - [ ] Code is clear and concise
   - [ ] UI code is internationalised
   - [ ] Code has adequate comments where appropriate
   - Code is in an appropriate place
     - [ ] Classes are in appropriate packages
     - [ ] Code fits with class's responsibilities (SRP)
   - [ ] Configuration files are documented enough
   - If code is removed
     - [ ] No remaining code references the removed code
       - [ ] No orphaned rewrite rules
       - [ ] No orphaned config settings
     - [ ] No remaining comments refer to the removed code
     - [ ] Unused imports and libraries are removed
 - Testing
   - [ ] New code has tests
   - [ ] Changed code has tests
   - [ ] All tests pass
   - [ ] Test coverage stable or increased
 - Documentation
   - [ ] New features are documented
   - [ ] Docs removed for deleted features
   - [ ] Docs updated for all changed features
   - [ ] Developer/sysadmin docs updated if appropriate
 - If there are new dependencies
   - [ ] Does each new dependency pass all the
         [Considerations for new dependencies](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines#new-technologies-and-dependencies)?


----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zanata/zanata-platform/408)
<!-- Reviewable:end -->

aeng commented:
Reviewed 2 of 4 files at r1.
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 46 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-KngsAR26oXKGrj00ZvY:-KngsAR3yp7J5BbGSfQ6:b-d0kko9) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L46)):*
> ```JavaScript
> 
> SettingsOptions.propTypes = {
>   states: PropTypes.shape({
> ```

Its better to use some sort of enum or list of static object(id, description) as key for the states.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 4 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngr_fghCHFkjqcwMrQ:-Kngr_fghCHFkjqcwMrR:b-289ad9) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L4)):*
> ```JavaScript
> import PropTypes from 'prop-types'
> import { storiesOf, action } from '@kadira/storybook'
> import RealSettingsOptions from '.'
> ```

Is our component name 'ReadSettingsOptions'?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 46 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-KngsAR26oXKGrj00ZvY:-KngvEDrl_xOFpOe4-nR:b-lvsgh8) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L46)):*
<details><summary><i>Previously, aeng (Alex Eng) wrote…</i></summary><blockquote>

Its better to use some sort of enum or list of static object(id, description) as key for the states.
</blockquote></details>

@davidmason  created the ValidationOptions this way, just followed what he did

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 4 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngr_fghCHFkjqcwMrQ:-KngvNXW2GjlzmBIvEV4:bhgxvll) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L4)):*
<details><summary><i>Previously, aeng (Alex Eng) wrote…</i></summary><blockquote>

Is our component name 'ReadSettingsOptions'?
</blockquote></details>

@davidmason used this approach on the ValidationOptions story - there is some reason for doing it but I am not qualified enough to explain what that is. It has something to do with creating the states for the checkboxes I believe.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 46 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-KngsAR26oXKGrj00ZvY:-KngykBBQsprzGZtzz1n:b-5b3i37) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L46)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

@davidmason  created the ValidationOptions this way, just followed what he did
</blockquote></details>

Ok, I have had this explained to me but I am unsure how to do this. When I do, I will also update the ValidationOptions story as it has the same issue. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 46 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-KngsAR26oXKGrj00ZvY:-Knh1rRaFp46sEkg_34S:b-py6udq) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L46)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

Ok, I have had this explained to me but I am unsure how to do this. When I do, I will also update the ValidationOptions story as it has the same issue. 
</blockquote></details>

- We should have one `SettingOption` component that displays a setting checkbox that takes props like:

```
setting: PropTypes.shape({
  id: PropTypes.any.isRequired, // I will update this to whatever I use when I wire it up
  label: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
}).isRequired
/* arguments: (any: settingId, bool: active) */
updateSetting: PropTypes.func.isRequired
```

- We should have one `SettingsOptions` component responsible to display a list of settings. Its props would just be a list of the setting option data type

```
settings: PropTypes.arrayOf(shape(
  id: PropTypes.any.isRequired, // I will update this to whatever I use when I wire it up
  label: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
)).isRequired
/* arguments: (any: settingId, bool: active) */
updateSetting: PropTypes.func.isRequired
```

- We should use the `SettingsOptions` anywhere we need a list of settings.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 46 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-KngsAR26oXKGrj00ZvY:-Knh4ZNWwxwYcqdCMs-J:btrxzwm) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L46)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

- We should have one `SettingOption` component that displays a setting checkbox that takes props like:

```
setting: PropTypes.shape({
  id: PropTypes.any.isRequired, // I will update this to whatever I use when I wire it up
  label: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
}).isRequired
/* arguments: (any: settingId, bool: active) */
updateSetting: PropTypes.func.isRequired
```

- We should have one `SettingsOptions` component responsible to display a list of settings. Its props would just be a list of the setting option data type

```
settings: PropTypes.arrayOf(shape(
  id: PropTypes.any.isRequired, // I will update this to whatever I use when I wire it up
  label: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
)).isRequired
/* arguments: (any: settingId, bool: active) */
updateSetting: PropTypes.func.isRequired
```

- We should use the `SettingsOptions` anywhere we need a list of settings.
</blockquote></details>

Thanks for that. I shall rewrite the SettingOptions component and remove the ValidationOptions component since it will the new component. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 4 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngr_fghCHFkjqcwMrQ:-KnhAWwyXECINFfKJnuQ:bihfd7n) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L4)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

@davidmason used this approach on the ValidationOptions story - there is some reason for doing it but I am not qualified enough to explain what that is. It has something to do with creating the states for the checkboxes I believe.
</blockquote></details>

@aeng the idea here is:

 - We want the wrapper component so that you can use the checkboxes in storybook
 - Checkbox state is managed in the redux app, but we don't want to run the whole app in storybook, so the wrapper component just manages it for the storybook
 - We want the code in the stories below to look exactly how you will use it in the app, because it will be displayed by one of the storybook plugins
 - We name the wrapper component the same as the real component so it looks the same in the code, and we import the real component as Real* to keep the name available for the wrapper.

For next tech-debt sprint I want to update to webpack 2, update storybook to latest and turn on a bunch of storybook plugins. Once that is done we will see the benefit of using `RealMyComponent` and wrapper `MyComponent`, instead of `MyComponent` and `Wrapper`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 4 at r1](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngr_fghCHFkjqcwMrQ:-KnhCEynXYD0FADAKvCs:b-te9glq) ([raw file](https://github.com/zanata/zanata-platform/blob/37705c9335af3d448008245590a24ebbca947bef/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L4)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

@aeng the idea here is:

 - We want the wrapper component so that you can use the checkboxes in storybook
 - Checkbox state is managed in the redux app, but we don't want to run the whole app in storybook, so the wrapper component just manages it for the storybook
 - We want the code in the stories below to look exactly how you will use it in the app, because it will be displayed by one of the storybook plugins
 - We name the wrapper component the same as the real component so it looks the same in the code, and we import the real component as Real* to keep the name available for the wrapper.

For next tech-debt sprint I want to update to webpack 2, update storybook to latest and turn on a bunch of storybook plugins. Once that is done we will see the benefit of using `RealMyComponent` and wrapper `MyComponent`, instead of `MyComponent` and `Wrapper`.
</blockquote></details>

TLDR: we call the wrapper component `SettingsOptions` so we will have the right code in our storybook style guide, so we have to use a different name for the import of the real component.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

aeng commented:
Reviewed 1 of 4 files at r1.
Review status: 3 of 4 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
**_Storybook progress_**

SettingOption - one checkbox in storybook - unsure how to show labels
SettingsOptions - one checkbox in storybook - unsure how to populate list from the component

May need some help @davidmason in figuring out how to get the labels to show and the best way to populate the list of checkbox items with labels. Shall look again in the morning and reach out if I am truly stumped. Slowly making progress.

kathryngo commented on server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js:
```diff
@@ -0,0 +1,40 @@
+import SettingOption from '../SettingOption'
+
+const SettingsOptions = ({states, updateSettingOption}) => {
+  const settings =
```
Unsure how to set default props for the nested props under settings and what their default values would be. I know what I have included here is incorrect, just was using it to test how storybook looks.

kathryngo commented on server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js:
```diff
@@ -0,0 +1,51 @@
+}
+
+const updateAction = action('updateSettingOption')
+const settings =
```
Know these are incorrect and need to be removed but I wanted to fix the prop code in the related index JS file first.

kathryngo commented on server/zanata-frontend/src/frontend/app/editor/containers/Sidebar/index.css:
```diff
@@ -380,6 +380,17 @@ h2.validation {
   color: var(--Sidebar-color-text);
 }
 
+.settings-options h2 {
```
Shall remove this css - was for headings which wont be included in this component

kathryngo commented on server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js:
```diff
@@ -0,0 +1,40 @@
+        'Printf variables',
+        'Tab characters (t)',
+        'XML entity reference']
+  const checkboxes = settings.map((settings, index) => (
```
should that be settings.map((**states**, index) ?

kathryngo commented on server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js:
```diff
@@ -0,0 +1,40 @@
+import PropTypes from 'prop-types'
+import SettingOption from '../SettingOption'
+
+const SettingsOptions = ({states, updateSettingOption}) => {
```
const correct?
Had no luck with classes

davidmason commented:
Review status: 1 of 6 files reviewed at latest revision, 6 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 5 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngq26LCohB6Nv3k4_--r3-5:-KnnnXbB5p-FvPGveqS0:b-adva4r) ([raw file](https://github.com/zanata/zanata-platform/blob/9c62880a09221fff5d0e86b686d14b419307d8f3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L5)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

const correct?
Had no luck with classes
</blockquote></details>

`const` is right for a stateless component. The setting checkbox component should be handling the state so it should be good here.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 6 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngq26LCohB6Nv3k4_--r3-6:-Knnnou97bIB35Jm-CuQ:b6phxe3) ([raw file](https://github.com/zanata/zanata-platform/blob/9c62880a09221fff5d0e86b686d14b419307d8f3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L6)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

Unsure how to set default props for the nested props under settings and what their default values would be. I know what I have included here is incorrect, just was using it to test how storybook looks. 

I am also unsure about how to pull in the states from SettingOption
</blockquote></details>

Because a stateless component is just a render function, you have to add `propTypes` and `defaultProps` after:

```
const MyComponent = ({a, b}) => <span>{a} {b}</span>

MyComponent.propTypes = {
  a: PropTypes.string.isRequired,
  b: PropTypes.string.isRequired,
}

MyComponent.defaultProps = {
  a: 'like',
  b: 'this'
}
```

But let's sort this out with a screenshare, sounds like there is too much for me to cover with typing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Review status: 1 of 6 files reviewed at latest revision, 6 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 14 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngq26LCohB6Nv3k4_--r3-14:-KnnsHzJP0MWASD1JnCo:b-4qz4v) ([raw file](https://github.com/zanata/zanata-platform/blob/9c62880a09221fff5d0e86b686d14b419307d8f3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L14)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

should that be settings.map((**states**, index) ?
</blockquote></details>

It can be called anything, but the most sensible thing here would be `setting`.

I would avoid calling things `state` or `states` because it is way too general, does not give any idea what it is (the exceptions in our apps are when it is the redux state, so in mapStateToProps and in the reducers, but that's it).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 1 of 6 files reviewed at latest revision, 6 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js, line 6 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngq26LCohB6Nv3k4_--r3-6:-KnqIAJT-C6q1YmHl3Lf:brvl1tx) ([raw file](https://github.com/zanata/zanata-platform/blob/9c62880a09221fff5d0e86b686d14b419307d8f3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/index.js#L6)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Because a stateless component is just a render function, you have to add `propTypes` and `defaultProps` after:

```
const MyComponent = ({a, b}) => <span>{a} {b}</span>

MyComponent.propTypes = {
  a: PropTypes.string.isRequired,
  b: PropTypes.string.isRequired,
}

MyComponent.defaultProps = {
  a: 'like',
  b: 'this'
}
```

But let's sort this out with a screenshare, sounds like there is too much for me to cover with typing.
</blockquote></details>

Understood. Will arrange a time for us to go through this over screenshare today

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Reviewed 1 of 4 files at r1, 1 of 5 files at r2, 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 27 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko5M18T9wP_3IlSWyJw:-Ko5M18UHYAJIjBJG8QR:blbr3fb) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L27)):*
> ```JavaScript
>         <RealSettingOption
>             updateSetting={this.updateSetting}
>             states={this.state}
> ```

This line looks like it is here by accident. Delete it.

---

*[server/zanata-frontend/src/frontend/app/editor/containers/Sidebar/index.css, line 383 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngq26LCohB6Nv3k4_1-r3-383:-Ko5O0PjB2ag-mYsMINx:b-csepva) ([raw file](https://github.com/zanata/zanata-platform/blob/9c62880a09221fff5d0e86b686d14b419307d8f3/server/zanata-frontend/src/frontend/app/editor/containers/Sidebar/index.css#L383)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

Shall remove this css - was for headings which wont be included in this component
</blockquote></details>

Looks like this is yet to be removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 27 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko5M18T9wP_3IlSWyJw:-Ko5OaHJZ_7m-6wrzlfo:buz22mh) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L27)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

This line looks like it is here by accident. Delete it.
</blockquote></details>

This PR is still labelled work in progress. I have not finished it yet so please wait before continuing your review. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 27 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko5M18T9wP_3IlSWyJw:-Ko5QpAZyHe89lHvMqy9:b-nnc42g) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L27)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

This PR is still labelled work in progress. I have not finished it yet so please wait before continuing your review. 
</blockquote></details>

Ah, whoops. Already reviewed everything here so far. That just means quicker review for the finished PR :grin: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

aeng commented:
Reviewed 1 of 4 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

aeng commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-:-Ko5RcjEVNNAn5KfxYwo:bnfp4nl)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 3 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 27 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko5M18T9wP_3IlSWyJw:-Ko5mkSM8T1isEN8CWPn:b-xtrnhj) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L27)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Ah, whoops. Already reviewed everything here so far. That just means quicker review for the finished PR :grin: 
</blockquote></details>

Everything has been fixed now. Ready for your review when you are ready.

---

*[server/zanata-frontend/src/frontend/app/editor/containers/Sidebar/index.css, line 383 at r3](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Kngq26LCohB6Nv3k4_1-r3-383:-Ko5mp9tWe1FDllJG2sj:b-66omwu) ([raw file](https://github.com/zanata/zanata-platform/blob/9c62880a09221fff5d0e86b686d14b419307d8f3/server/zanata-frontend/src/frontend/app/editor/containers/Sidebar/index.css#L383)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Looks like this is yet to be removed.
</blockquote></details>

I needed to keep this css and add a bit more - for the header styling and spacing between lists

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Reviewed 6 of 6 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 27 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko5M18T9wP_3IlSWyJw:-Ko61uiDsctmsz3goTBY:banck67) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L27)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

Everything has been fixed now. Ready for your review when you are ready.
</blockquote></details>

You deleted the wrapper component from this storybook. It is now not possible to check/uncheck the checkbox in storybook. Is that what you intended?

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 6 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko62Aqzu1AR8MJ5t7Tv:-Ko62Aqzu1AR8MJ5t7Tw:b-nx1vsr) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L6)):*
> ```JavaScript
> import RealSettingsOptions from '.'
> 
> class SettingsOptions extends React.Component {
> ```

You deleted this wrapper component, so the checkboxes will not work in storybook. Is that what you want here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 6 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko62Aqzu1AR8MJ5t7Tv:-Ko62xVtaPuVwujn2osG:b-z7elct) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L6)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

You deleted this wrapper component, so the checkboxes will not work in storybook. Is that what you want here?
</blockquote></details>

The checkboxes do work in storybook after deleting this wrapper. After removing the Real... component code and its import statement it worked so I thought this is what I needed to do

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Review status: 6 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 6 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko62Aqzu1AR8MJ5t7Tv:-Ko6CMXxwyvFVXR_VBmx:b9m51ul) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L6)):*
<details><summary><i>Previously, kathryngo wrote…</i></summary><blockquote>

The checkboxes do work in storybook after deleting this wrapper. After removing the Real... component code and its import statement it worked so I thought this is what I needed to do
</blockquote></details>

Ok, so that means react-bootstrap checkboxes must not be a controlled component. I'll need to look into what the deal is there, since we want controlled components to avoid problems where the component shows the wrong state compared to the app state. Don't worry about that for now though, I'll look into it when wiring it up.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 5 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 27 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko5M18T9wP_3IlSWyJw:-Ko6F07HRr3IdIKbtpZh:b-886gsx) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L27)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

You deleted the wrapper component from this storybook. It is now not possible to check/uncheck the checkbox in storybook. Is that what you intended?
</blockquote></details>

Fixed

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js, line 6 at r4](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko62Aqzu1AR8MJ5t7Tv:-Ko6F2q7B5F6s9lt7Y7h:b-886gsx) ([raw file](https://github.com/zanata/zanata-platform/blob/fe2d165fa81a5783e818b274efb74597f9c8bef3/server/zanata-frontend/src/frontend/app/editor/components/SettingsOptions/SettingsOptions.story.js#L6)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

Ok, so that means react-bootstrap checkboxes must not be a controlled component. I'll need to look into what the deal is there, since we want controlled components to avoid problems where the component shows the wrong state compared to the app state. Don't worry about that for now though, I'll look into it when wiring it up.
</blockquote></details>

Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Reviewed 2 of 3 files at r6.
Review status: 7 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 34 at r6](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko6RPxr_Wyvksuqs9ul:-Ko6RPxr_Wyvksuqs9um:blxjm54) ([raw file](https://github.com/zanata/zanata-platform/blob/a6c82ecc3da6ae19de8287d84f635e7b0f63140a/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L34)):*
> ```JavaScript
> }
> 
> const updateSetting = action('updateSetting')
> ```

This new line of code `const updateSetting = action('updateSetting')` looks unused.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

kathryngo commented:
Review status: 6 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js, line 34 at r6](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-Ko6RPxr_Wyvksuqs9ul:-Ko6UVyqDxQgxbbk6Ojh:b-886gsx) ([raw file](https://github.com/zanata/zanata-platform/blob/a6c82ecc3da6ae19de8287d84f635e7b0f63140a/server/zanata-frontend/src/frontend/app/editor/components/SettingOption/SettingOption.story.js#L34)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary><blockquote>

This new line of code `const updateSetting = action('updateSetting')` looks unused.
</blockquote></details>

Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->

davidmason commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 3 files at r6, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408#-:-Ko6VViWLjh1UsZsMcVH:bnfp4nl)*
<!-- Sent from Reviewable.io -->

davidmason commented:
Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-platform/408)*
<!-- Sent from Reviewable.io -->
